.github/workflows/ci.yaml

```diff
@@ -0,0 +1,155 @@
+name: Tupaia CI
+on:
+  push:
+    branches:
+      - '*'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  NODE_OPTIONS: '--max-old-space-size=6144'
+  NODE_ENV: test
+
+jobs:
+  validate_branch_name:
+    name: Validate branch name
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Validate branch name
+        run: packages/devops/scripts/ci/validateBranchName.sh
+
+  validate_migration:
+    name: Validate new migrations
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Validate new migration
+        run: packages/devops/scripts/ci/validateNewMigrations.sh
+
+  #    - TODO: use eslint no-exclusive-tests flag at lint step after RN-201
+  validate_tests:
+    name: Validate unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14.19.3
+          cache: yarn
+      - run: yarn set version 3.2.4
+      - run: SKIP_BUILD_INTERNAL_DEPENDENCIES=true yarn install --frozen-lockfile
+      - run: yarn workspace @tupaia/utils build # /scripts/node/validateTests.js use utils package
+
+      - name: Validate unit tests
+        run: packages/devops/scripts/ci/validateTests.sh
+
+  pull_secrets_from_lastpass:
+    name: Pull secrets from LastPass
+    runs-on: ubuntu-latest
+    outputs:
+      db_pg_password: ${{ steps.pull_env_vars.outputs.DB_PG_PASSWORD }}
+
+    steps:
+      - name: Install LastPass CLI
+        run: |
+          sudo apt-get update -y
+          sudo apt-get upgrade -y
+          sudo apt-get install -y lastpass-cli
+      - name: Login to LastPass
+        run: echo ${{ secrets.LASTPASS_PASSWORD }} | LPASS_DISABLE_PINENTRY=1 lpass login ${{ secrets.LASTPASS_USERNAME }}
+      - id: pull_env_vars
+        run: |
+          lpass show --notes "GitHub Action .env Environment Variables" >> secrets.env
+          cat secrets.env | grep DB_PG_PASSWORD >> "$GITHUB_OUTPUT"
+      - name: Upload secrets artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: secrets
+          path: secrets.env
```

Comment:
Here is to use Artifact to share secrets between jobs. I've tried using $GITHUB.OUTPUT and `::add-mask::` passing these secrets, but it is not satisfied or the secrets got expose.